### PR TITLE
Amend RFC0070 with DeclBuffer TVMScript syntax updates

### PR DESCRIPTION
**Pull request by vinx13**
As a followup on the previous discussion https://github.com/apache/tvm-rfcs/pull/70#issuecomment-1149308146, this PR updated the DeclBuffer RFC with TVMScript related changes.

cc @Lunderberg @wrongtest-intellif @Hzfengsy @junrushao1994 @cyx-6 @tqchen

### CODE REVIEW

**Review by Hzfengsy — APPROVED** [review]

File: rfcs/0070-introducing-decl-buffer.md [hunk 1]
```diff
@@ -146,9 +146,36 @@ Specifically, the updated flow of buffer flattening using `DeclBuffer` will be:
 with flattened indices.
 
 ## TVM script updates
+* New statement `T.decl_buffer` will be introduced. It has the same interface as `T.buffer_decl`.
+```python
+def decl_buffer(
+    shape: Sequence[Union[PrimExpr, int]],
+    dtype: str = "float32",
+    data: Var = None,
```
Comment (areusch) [inline rfcs/0070-introducing-decl-buffer.md:154]: just one note--usually this is first if positional args are used (e.g. in the example above), is the order of these kwargs correct?

**Review by areusch — APPROVED** [review]
thanks @vinx13 !

File: rfcs/0070-introducing-decl-buffer.md (hunk 1 above)
Comment (vinx13) [reply rfcs/0070-introducing-decl-buffer.md:154]: I'm keeping the order the same as the existing `T.buffer_decl`

Comment (areusch) [reply rfcs/0070-introducing-decl-buffer.md:154]: ok, but in the example above (line 135) you have:
```
A_flattened = T.decl_buffer(A.data, (256,), "float32")
```

this says to me that:
- `shape` == `A.data`
- `dtype` == `(256,)`
- `data` == `"float32"`

could you keep the RFC consistent?

Comment (vinx13) [reply rfcs/0070-introducing-decl-buffer.md:154]: Thank you for pointing out, I've updated the RFC

**Comment by areusch** [comment]
lgtm!

**Pull request merged**